ddtrace/_trace/span.py

```diff
@@ -634,6 +650,24 @@ def set_link(
             attributes=attributes,
         )
 
+    def set_pointer(
+        self,
+        kind: str,
+        direction: SpanPointerDirection,
+        hash: SpanPointerHash,
+        **extra_attributes: Any,
+    ) -> None:
+        pointers_key = PointersKey(hash, direction)
+
+        if pointers_key in self._pointers:
```

Comment:
this may actually not be necessary. though it may be strange to have multiple pointers to the same entity and direction, it's not necessarily a bug. so we might want to just keep track of the `SpanPointer`s as a list instead of a dictionary.